tests/kubevirt_helper.go

```diff
@@ -0,0 +1,200 @@
+package tests
+
+import (
+	context1 "context"
+	"fmt"
+	"github.com/portworx/sched-ops/k8s/core"
+	kubevirtdy "github.com/portworx/sched-ops/k8s/kubevirt-dynamic"
+	"github.com/portworx/sched-ops/task"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/volume"
+	"github.com/portworx/torpedo/pkg/log"
+	corev1 "k8s.io/api/core/v1"
+	"time"
+)
+
+// AddDisksToKubevirtVM is a function which takes number of disks to add and adds them to the kubevirt VMs passed (Please provide size in Gi)
+func AddDisksToKubevirtVM(virtualMachines []*scheduler.Context, numberOfDisks int, size string) (bool, error) {
+	// before adding disks check how many disks are present in the VM
+	for _, appCtx := range virtualMachines {
+		vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{appCtx})
+		if err != nil {
+			return false, err
+		}
+		for _, v := range vms {
+			t := func() (interface{}, bool, error) {
+				diskCountOutput, err := GetNumberOfDisksInVM(v)
+				if err != nil {
+					return nil, false, fmt.Errorf("failed to get number of disks in VM [%s] in namespace [%s]", v.Name, v.Namespace)
+				}
+				// Total disks will be numberOfVolumes plus the container disk
+				if diskCountOutput == 0 {
+					return nil, true, fmt.Errorf("expected number of disks in VM [%s] in namespace [%s] is [%d] but got [%d]", v.Name, v.Namespace, numberOfDisks+1, diskCountOutput)
+				}
+				return diskCountOutput, false, nil
+			}
+			d, err := task.DoRetryWithTimeout(t, 10*time.Minute, 30*time.Second)
+			if err != nil {
+				return false, err
+			}
+			diskCount := d.(int)
+			log.InfoD("Number of disks in VM [%s] in namespace [%s] is [%d]", v.Name, v.Namespace, diskCount)
+
+			// Before we add the pvc we need to get storage class of the pvc
+			storageClass, err := GetStorageClassOfVmPVC(appCtx)
+			if err != nil {
+				return false, err
+			}
+			log.InfoD("Storage class of PVC attached to VM [%s] in namespace [%s] is [%s]", v.Name, v.Namespace, storageClass)
+
+			// Add the disks to the VM
+			pvcs, err := CreatePVCsForVM(v, 1, storageClass, size)
+
+			if err != nil {
+				return false, err
+			}
+
+			specListInterfaces := make([]interface{}, len(pvcs))
+			for i, pvc := range pvcs {
+				// Converting each PVC to interface for appending to SpecList
+				specListInterfaces[i] = pvc
+			}
+			appCtx.App.SpecList = append(appCtx.App.SpecList, specListInterfaces...)
+
+			err = AddPVCsToVirtualMachine(v, pvcs)
+			if err != nil {
+				return false, err
+			}
+
+			err = RestartKubevirtVM(v.Name, v.Namespace, true)
+			if err != nil {
+				return false, err
+			}
+			log.InfoD("Sleep for 5mins for vm to come up")
+			time.Sleep(5 * time.Minute)
```

Comment:
Note - in the next PR pls remove this hard sleep and work with ssh method instead.